Lib/idlelib/autocomplete.py

```diff
@@ -106,7 +272,17 @@ def open_completions(self, args):
         hp = HyperParser(self.editwin, "insert")
         curline = self.text.get("insert linestart", "insert")
         i = j = len(curline)
-        if hp.is_in_string() and (not mode or mode==FILES):
+
+        if mode in (None, DICTKEYS):
```

Comment:
It looks like this suite should be in the block for `elif mode in (None, DICTKEYS):` (revised from  `elif mode == DICTKEYS:` ).

Reply:
I'm not sure what you mean... I put this block at the top intentionally. Note that it is followed by `if` rather than `elif`, so that if the block doesn't decide that dict key completion should take place, the other completion modes are checked.

Reply:
I did notice, but it is not yet obvious why the preliminary check in needed if it can be overriden.  But leave it for now.